hatchways/team-pine
# Setup forgot password forms and integrate with SendGrid / Back end

### What this PR does:
- This PR adds the forgot password forms, both send password reset email form and the password reset form.
- All functionality is present on the back end to send an email to requested email for a reset (if it exists in the user database), generate a link with a token along with a hashed version of the token in the database that expires after 30 minutes, and match the tokens later on password reset to verify the same user who has the email is the one who entered the new password.
- As a side effect of making the auth page wrapper responsive, the login page and signup page are now properly responsive.

### Screenshots / Videos:
![image](https://user-images.githubusercontent.com/34494229/154082606-06b25811-748e-4b94-ac64-d8d21ccad59b.png)
![image](https://user-images.githubusercontent.com/34494229/154082756-b118285d-1de8-4668-8f7a-aabe4840137a.png)
![image](https://user-images.githubusercontent.com/34494229/154084378-3701f2fd-cb39-464e-b1b1-4ccbaf15e200.png)
![image](https://user-images.githubusercontent.com/34494229/154084533-04de7314-06a9-46ea-b786-863e1e2b3b83.png)
![image](https://user-images.githubusercontent.com/34494229/154084827-69a404c6-0149-4cd0-a030-55463de65dae.png)
![image](https://user-images.githubusercontent.com/34494229/154084982-281adb79-fc95-47df-8486-104bf1a6780b.png)
![image](https://user-images.githubusercontent.com/34494229/154085147-70d67db2-7a4e-43fc-b480-a17302b4396e.png)

### Any information needed to test this feature:
- Request the API key for your .env file via a secure channel
- Click on forgot password link and enter an email that is both linked to an account and accessible.
- Once you click on the link that was emailed to you, enter the new password and confirm it.
- If successful, you should be redirected to the login page. Enter the new password to test.
- To test incorrect entries:
1. Try entering your own link with an incorrect token (/password-reset/:email/:token) and try changing the password.
2. Enter an email that is does not exist under a user.
3. Try using a token after 30 minutes have passed.
4. Try submitting with no password or unmatched passwords.

### Any issues with the current functionality:
- There is no password history functionality.
- The support email mentioned in the reset email is not actually implemented.

closes #84
